### PR TITLE
Discount code widget

### PR DESCRIPTION
**Pull request by rcrphillips**
## What does this change?
In order to fill our contractual obligation for the discount code site, editorial have selected ~250 articles to display the new widget. I've placed it as low as possible on the page. Design via @zeftilldeath and Scala help from @philmcmahon 
## Screenshots
Screen Shot 2019-06-06 at 20 21 36

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: applications/app/views/fragments/indexBody.scala.html [hunk 1]
```diff
@@ -8,6 +8,7 @@
 @import services.IndexPage
 @import views.html.fragments.containers.facia_cards.{container, containerScaffold, audioFlagshipContainer}
 @import views.html.fragments.pagination
+@import views.support.Commercial
```
Comment (philmcmahon) [inline applications/app/views/fragments/indexBody.scala.html]: this is probably unncessary :)

File: common/app/common/commercial/DiscountCodeWidget.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Amazon,
+    Currys,
+    JdSports
+  )
+
+  def shouldShowWidget(path:String): Boolean ={
+    widgetPaths.contains(path)
+  }
+
+  val widgetPaths = Set(
```
Comment (philmcmahon) [inline common/app/common/commercial/DiscountCodeWidget.scala]: this could be a private val maybe

**Review by philmcmahon — APPROVED** [review]
Couple of tiny things otherwise think this is good to go!

File: common/app/common/commercial/DiscountCodeWidget.scala (hunk 2 above)
Comment (rcrphillips) [reply common/app/common/commercial/DiscountCodeWidget.scala]: Fixed

File: applications/app/views/fragments/indexBody.scala.html (hunk 1 above)
Comment (rcrphillips) [reply applications/app/views/fragments/indexBody.scala.html]: Fixed

**Pull request merged**